### DISCUSSION TITLE
logging into backup server?

### DISCUSSION THREAD

**Discussion by nycalex** [jhuckaby/Cronicle#734] [General]
Hello again!

I can't seem to find documentation on logging into the backup server.  Should I be able to? i know workers i can't login to...

In the primary i can see that it is successfully communicating to the backup. But the backup doesn't seem to accept any passwords - not default one, not the password i set for the admin in the primary.

Thanks for your time
Cheers!

**Comment by jhuckaby** [comment]
Nope, the backup servers are designed to be totally dormant until they take over and become primary.  You cannot use them to login or anything when they are in backup mode.

Also, note that when using a multi-primary server cluster, Cronicle requires a shared storage system like S3, so all the servers can share the same data (i.e. users / sessions / schedule).  It sounds like you don't have this set up correctly.

**Comment by nycalex** [comment]
indeed, not using shared storage, just synced manually. 

One thing  is not clear in the documentation:  if there is an identical config.json file for both primary and backup, wouldn't that "break" the model of the specifying the each server's hostname in base_app_url ?

**Reply by jhuckaby** [reply]
> indeed, not using shared storage, just synced manually.

Okay, so just FYI, that's not a supported setup.  Cronicle requires [some kind of shared storage system](https://github.com/jhuckaby/Cronicle/blob/master/docs/Setup.md#multi-server-cluster) for multi-master setups.

> if there is an identical config.json file for both primary and backup, wouldn't that "break" the model of the specifying the each server's hostname in base_app_url ?

Good point, the `base_app_url` does need to be different, when you are not using a proxy or load balancer in front.  I'll add a note to the docs.  Thanks!

**Comment by nycalex** [comment]
now i'm even more confused :)
if i use shared media like an NFS mount, both primary and backup servers will share the config file and hence the base_app_url will be the same.  I'm probably missing something, but shared media for config file and host-specific base_app_url sound like mutually exclusive ideas...

**Reply by jhuckaby** [reply]
Apologies, my confusion.  You're using an NFS mount for the shared data?  Great!  You should be good there.

Sorry, I was trying to separate your question into two separate parts.  The shared storage with multi-master situation (should be fine now, thank you for the clarification), and separately, the config file and `base_app_url` property.  That DOES need to be separate, if you aren't using a load balancer or proxy.

I hope that clears up the confusion.  Sorry!

**Reply by nycalex** [reply]
i'm not using shared media, i was just bringing it up with an IF, as a hypothetical. 

so IF i use some shared media NFS or S3, wouldn't both servers end up sharing the config file including the base_app_url that's in the config json?  That's what i struggle to wrap around my head. Thanks again!!!

**Reply by jhuckaby** [reply]
Oh sorry!  I get it now.

> so IF i use some shared media NFS or S3, wouldn't both servers end up sharing the config file including the base_app_url that's in the config json?

Nope!  The config file is separate from the "data" storage.  It's just a file on disk (`/opt/cronicle/conf/config.json`).  So you can keep the two config files separate, each with its own separate and unique `base_app_url` for each of your servers, and both can be set to use the same shared storage.
